Ui ux improvements

damms005 opened the pull request:
Please reference [PR#42](https://github.com/nafiesl/free-pmo/pull/42)

nafiesl commented:
@damms005, looks like lot of file changes just for a permission change, I will push new commit to revert file permission.

nafiesl commented on config/app.php:
```diff
@@ -13,7 +13,7 @@
     */
 
-    'name' => 'Free PMO',
+    'name' => 'Anchis & Kolbins',
```
Please revert back this config `app.name` to **"Free PMO"**.

damms005 replied:
Done

nafiesl commented on resources/views/layouts/app.blade.php:
```diff
@@ -11,6 +11,15 @@
+
+    <style>
+
+.sidebar {
```
Please fix this indentation. This sould be like:
```html
    <style>
        .sidebar {
             margin-top: 0px;
        }
    </style>
```

nafiesl commented on resources/views/layouts/partials/lang-switcher.blade.php:
```diff
@@ -1,7 +1,8 @@
+    {{-- {{ trans('lang.lang') }} : --}}
 
     @foreach (['en', 'id', 'de'] as $langKey)
+        {{-- @break --}}
```
Remove this

damms005 replied:
@nafiesl  If I remove this, the GitHub auto Travis build for this project fails. The failure is in a UI test. Specifically, `UserProfileTest->user_can_switch_lang_from_sidebar()` test fails

nafiesl replied:
Ok, i will see what i can do to fix this.

nafiesl commented on resources/views/layouts/partials/lang-switcher.blade.php:
```diff
@@ -1,7 +1,8 @@
-<div class="text-center" style="border-bottom: 1px solid #e7e7e7; padding-bottom: 10px">
-    {{ trans('lang.lang') }} :
+<div class="text-center" style="border-bottom: 1px solid #e7e7e7; padding: 90px 10px 0px 0px">
+    {{-- {{ trans('lang.lang') }} : --}}
```
Why do you removing this?

damms005 replied:
Sorry, oversight :). Fixed!

nafiesl commented on routes/web.php:
```diff
@@ -1,5 +1,7 @@
 <?php
 
+//upload table issues
```
Remove this commented script.

nafiesl commented on app/Http/Requests/Projects/CreateRequest.php:
```diff
@@ -24,15 +24,25 @@ public function authorize()
+        //achieved using environmental variable.
+        //A value of zero (0) will mean "no limit"
+
+        $char_len_project_description = intval(env('CHAR_LEN_PROJECT_DESCRIPTION', 255));
```
I believe we just need 1 new env variable here, change it to just: **DESCRIPTION_CHAR_LEN**. Add this to **.env.example** file. And please use **camelCase** variable names.

nafiesl commented on app/Http/Requests/Projects/UpdateRequest.php:
```diff
@@ -38,5 +37,16 @@ public function rules()
+        //achieved using environmental variable.
+        //A value of zero (0) will mean "no limit"
+
+        $char_len_project_description = intval(env('CHAR_LEN_PROJECT_DESCRIPTION', 255));
```
Change this to **DESCRIPTION_CHAR_LEN**, too. And change variable name to **camelCase**.

nafiesl commented on app/Http/Requests/Tasks/CreateRequest.php:
```diff
@@ -24,10 +24,20 @@ public function authorize()
+        //achieved using environmental variable.
+        //A value of zero (0) will mean "no limit"
+
+        $char_len_task_description = intval(env('CHAR_LEN_TASK_DESCRIPTION', 255));
```
Change this to **DESCRIPTION_CHAR_LEN**, too. And change variable name to **camelCase**.

nafiesl commented on app/Http/Requests/Jobs/UpdateRequest.php:
```diff
@@ -26,12 +26,22 @@ public function authorize()
+        //achieved using environmental variable.
+        //A value of zero (0) will mean "no limit"
+
+        $char_len_job_description = intval(env('CHAR_LEN_JOB_DESCRIPTION', 255));
```
Change this to **DESCRIPTION_CHAR_LEN**, too. And change variable name to **camelCase**.

nafiesl replied:
This line has not been fixed. Please use DESCRIPTION_CHAR_LEN and **camelCase** variable name

nafiesl commented on app/Http/Requests/Tasks/UpdateRequest.php:
```diff
@@ -25,11 +25,21 @@ public function authorize()
+        //achieved using environmental variable.
+        //A value of zero (0) will mean "no limit"
+
+        $char_len_task_description = intval(env('CHAR_LEN_TASK_DESCRIPTION', 255));
```
Change this to **DESCRIPTION_CHAR_LEN**, too. And change variable name to **camelCase**.

nafiesl commented on composer.json:
```diff
@@ -7,12 +7,13 @@
         "php": "^7.1.3",
         "backup-manager/laravel": "^1.2",
-        "laracasts/presenter": "^0.2.1",
+        "doctrine/dbal": "^2.9",
```
Move this `doctrine/dbal` dependency to `require-dev`, since is only used for sqlite, expecially just for running tests.

nafiesl commented:
wait @damms005, please do not push any other feature than UI and UX improvement.

I you want to add new feature like "jobs can now have file attachments", **please create another PR**, we will review that together.

nafiesl commented on composer.json:
```diff
@@ -7,16 +7,18 @@
         "riskihajar/terbilang": "^1.2",
-        "spatie/laravel-fractal": "^5.0"
+        "spatie/laravel-fractal": "<5.5",
+        "spatie/laravel-medialibrary": "^7.0.0"
```
I do not want to use this Package for file upload. please use regular file upload and create another PR for job attachment feature.

nafiesl commented on config/app.php:
```diff
@@ -13,7 +13,7 @@
     */
 
-    'name' => 'Free PMO',
+    'name' => 'FREE PMO',
```
**Free** PMO, not FREE PMO.

nafiesl commented on resources/views/layouts/app.blade.php:
```diff
@@ -30,13 +40,14 @@
 
     <script type="text/javascript">
-    (function() {
+        (function() {
```
Fix indentation

damms005 commented:
> wait @damms005, please do not push any other feature than UI and UX improvement.
> 
> I you want to add new feature like "jobs can now have file attachments", **please create another PR**, we will review that together.

My bad! It was intended for another PR actually. Fixed!

nafiesl commented:
@damms005, for job attachment feature, it;s better if you craete that after we finish this PR.

damms005 commented:
@nafiesl  Please check the Travis build. Tests are failing arbitrarily. e.g. `user_can_login_and_logout` fails but there is nothing wrong with that feature (I did not work on that feature though). Also, almost all the tests on `events` (i.e. `/tests/Feature/Api/ApiEventsTest.php`) failed

Thanks.

nafiesl commented:
OK @damms005, i will check on travis in next couple of hours. And get back to you.

nafiesl commented:
@damms005, i believe this fails because you accidentally remove `spatie/laravel-fractal` from composer.json. We still use it on API endpoint features.

https://github.com/nafiesl/free-pmo/pull/43/files#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780L15

nafiesl commented:
@damms005 please fix your **indentation** on composer.json on last commit. And also you need to **update composer.lock** after that. As you can see, travis still fails.

ad3n commented:
@nafiesl can i get preview before merge this PR? very interesting to know

nafiesl commented:
Sure @ad3n, glad you can help :grin:

nafiesl commented on resources/views/layouts/app.blade.php:
```diff
@@ -30,26 +40,35 @@
+            max: 1024, //just an arbitrary default
+        	stopInputAtMaximum: false
         });
-    })();
```
@damms005 please do not remove this parentheses `()`. we use that for a reason.

nafiesl replied:
No problem @damms005, no need a hurry. We can continue in our free time.

damms005 commented:
@nafiesl It seems something is off with the config/setup for Travis. All PHPUnit tests passed on my system but now it's failing when I push same.

nafiesl commented:
Hi @damms005, let me review your latest update.

nafiesl commented:
@damms005 I have seen UI from this PR on my localhost, and get this dashboard view.

![screenshot_19-10-18_005](https://user-images.githubusercontent.com/8721551/67101173-b656ce80-f1f3-11e9-8f32-b4af982aef40.jpg)

Since this is UI improvement, I don't see this is better then existing one.

![screenshot_19-10-18_006](https://user-images.githubusercontent.com/8721551/67101073-84de0300-f1f3-11e9-9392-709f4e51c004.jpg)

Would you mind to explain this @damms005?

Besides, I see you include some css that we don't use.

damms005 commented:
@nafiesl:

1. Lots of changes that I cannot start listing here, but I'll suggest you `diff` commit `03788889379dd01d7f298b37487967de7b51bf98`. I removed a number of hard-coded stuff and made those more flexible, and added character counts
2. added remember me feature

nafiesl commented:
@damms005 I cannot review this PR, we can see too many change files here.
https://github.com/nafiesl/free-pmo/pull/43/files

I am suggesting you create other PR for each individual feature you want to contribute.

For example:
1. Create 1 PR for "remember me" feature
2. Create 1 PR for "other features" you want to contribute.

TBH, I don't like this UI improvement you give on this PR. I also don't want to add `spatie/laravel-fractal` package.

I am closing this PR, sorry.
